extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -129,46 +152,63 @@ private static QuarkusSecurityIdentity augmentIdentity(SecurityIdentity security
         return cf;
     }
 
-    private void processSuccessfulAuthentication(RoutingContext context, CompletableFuture<SecurityIdentity> cf,
+    private void processSuccessfulAuthentication(RoutingContext context,
+            CompletableFuture<SecurityIdentity> cf,
             AccessToken result, SecurityIdentity securityIdentity) {
-        removeSessionCookie(context);
+        removeCookie(context, SESSION_COOKIE_NAME);
         CookieImpl cookie = new CookieImpl(SESSION_COOKIE_NAME, new StringBuilder(result.opaqueIdToken())
-                .append(SESSION_COOKIE_DELIM)
+                .append(COOKIE_DELIM)
                 .append(result.opaqueAccessToken())
-                .append(SESSION_COOKIE_DELIM)
+                .append(COOKIE_DELIM)
                 .append(result.opaqueRefreshToken()).toString());
 
         cookie.setMaxAge(result.idToken().getInteger("exp"));
         cookie.setSecure(context.request().isSSL());
         cookie.setHttpOnly(true);
-
         context.response().addCookie(cookie);
+
         cf.complete(augmentIdentity(securityIdentity, result.opaqueAccessToken(),
                 result.opaqueRefreshToken()));
     }
 
-    private String generateState(RoutingContext context) {
-        CookieImpl cookie = new CookieImpl(STATE_COOKIE_NAME, UUID.randomUUID().toString());
+    private String getDynamicPath(RoutingContext context, URI absoluteUri) {
+        if (config.authentication.redirectPath.isPresent()) {
+            String redirectPath = config.authentication.redirectPath.get();
+            String requestPath = absoluteUri.getRawPath();
+            if (requestPath.startsWith(redirectPath) && requestPath.length() > redirectPath.length()) {
+                return requestPath.substring(redirectPath.length());
+            }
+        }
+        return null;
+    }
+
+    private String generateState(RoutingContext context, String dynamicPath) {
+        String uuid = UUID.randomUUID().toString();
+        String cookieValue = uuid + COOKIE_DELIM + dynamicPath;
+
+        CookieImpl cookie = new CookieImpl(STATE_COOKIE_NAME, cookieValue);
 
         cookie.setHttpOnly(true);
         cookie.setSecure(context.request().isSSL());
-        cookie.setMaxAge(-1);
+        cookie.setMaxAge(60 * 5);
```

Comment:
Is this really necessary given that we always reset the cookie when doing challenges? Assuming that the user may take more time than usual to log in at the AS, the cookie will be automatically removed and the redirect will fail.

Reply:
@pedroigor I updated it because when a `security-oidc-web-app-authentication-quickstart` is built a lot of warnings is displayed about an invalid `max-age=-1` cookie parameter :-). I set to to 5 mins, but we can set it to 30 mins, that should be enough, what do you think ?

Reply:
It should be fine, maybe we can also add an additional step after successfully obtaining the ID Token and remove the cookie ?

Reply:
@pedroigor yes, it is removed in this PR, but before the code step is started but I'll tweak it

Reply:
It is now removed only after the local redirect